src/safeguard-ps.psm1

```diff
@@ -1131,6 +1152,12 @@ function Invoke-SafeguardMethod
     $ErrorActionPreference = "Stop"
     if (-not $PSBoundParameters.ContainsKey("Verbose")) { $VerbosePreference = $PSCmdlet.GetVariableValue("VerbosePreference") }
 
+    if (-not ($PSBoundParameters.ContainsKey("Version") -and $SafeguardSession))
```

Comment:
Version will always have a value, since it's default value is 3. SafeguardSession is a variable initialized by Connect--safegaurd? won't it always be true? Can you please give an example of a case when we would get inside the if block.

Reply:
@Goelrachit 
What $PSBoundParameters evaluates is if the caller actually passed in the parameter.  It is false if it is just assigned to default, because the parameter wasn't "bound".

Although I think I see an error with the parentheses.  I tested it and it worked, so I need to check something.  Don't merge yet.

Reply:
Ahh ok